flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -968,6 +982,75 @@ public void testExceptionHistoryWithRestartableFailure() {
+                        TEMPORARY_FOLDER.newFile(), savepointId, operatorID);
+
+        // set savepoint settings which don't allow non restored state
+        final SavepointRestoreSettings savepointRestoreSettings =
+                SavepointRestoreSettings.forPath(savepointFile.getAbsolutePath(), false);
+
+        // create a new operator
+        final JobVertex jobVertex = new JobVertex("New operator");
+        jobVertex.setInvokableClass(NoOpInvokable.class);
+        final JobGraph jobGraphWithNewOperator =
+                TestUtils.createJobGraphFromJobVerticesWithCheckpointing(
+                        savepointRestoreSettings, jobVertex);
+
+        // creating the DefaultScheduler should try to restore the ExecutionGraph
+        SchedulerTestingUtils.newSchedulerBuilder(
+                        jobGraphWithNewOperator,
+                        ComponentMainThreadExecutorServiceAdapter.forMainThread())
+                .build();
+    }
+
+    @Test
+    public void testRestoringModifiedJobFromSavepointWithAllowNonRestoredStateSucceeds()
+            throws Exception {
+        // create savepoint data
+        final long savepointId = 42L;
+        final OperatorID operatorID = new OperatorID();
+        final File savepointFile =
+                TestUtils.createSavepointWithOperatorState(
+                        TEMPORARY_FOLDER.newFile(), savepointId, operatorID);
+
+        // allow for non restored state
+        final SavepointRestoreSettings savepointRestoreSettings =
+                SavepointRestoreSettings.forPath(savepointFile.getAbsolutePath(), true);
+
+        // create a new operator
+        final JobVertex jobVertex = new JobVertex("New operator");
+        jobVertex.setInvokableClass(NoOpInvokable.class);
+        final JobGraph jobGraphWithNewOperator =
+                TestUtils.createJobGraphFromJobVerticesWithCheckpointing(
+                        savepointRestoreSettings, jobVertex);
+
+        final StandaloneCompletedCheckpointStore completedCheckpointStore =
+                new StandaloneCompletedCheckpointStore(1);
+        final CheckpointRecoveryFactory testingCheckpointRecoveryFactory =
+                useSameServicesForAllJobs(
+                        completedCheckpointStore, new StandaloneCheckpointIDCounter());
+
+        SchedulerTestingUtils.newSchedulerBuilder(
+                        jobGraphWithNewOperator,
+                        ComponentMainThreadExecutorServiceAdapter.forMainThread())
+                .setCheckpointRecoveryFactory(testingCheckpointRecoveryFactory)
+                .build();
+
+        // creating the DefaultScheduler should have read the savepoint
+        final CompletedCheckpoint savepointCheckpoint =
+                completedCheckpointStore.getLatestCheckpoint(false);
+
+        MatcherAssert.assertThat(savepointCheckpoint, notNullValue());
+
+        MatcherAssert.assertThat(savepointCheckpoint.getCheckpointID(), is(savepointId));
```

Comment:
```suggestion
        final CompletedCheckpoint savepoint =
                completedCheckpointStore.getLatestCheckpoint(false);

        MatcherAssert.assertThat(savepoint, notNullValue());

        MatcherAssert.assertThat(savepoint.getCheckpointID(), is(savepointId));
```
I would remove the "Checkpoint" part here. At least to me, it was a bit confusing: Is it a checkpoint or a savepoint?

Reply:
This applies to the `AdaptiveSchedulerTest` accordingly.

Reply:
Will change it.